### PR TITLE
Video upload

### PR DESCRIPTION
**Pull request by mplorentz**
#934. This updates our image picker and Nostr.build API code to allow the user to upload videos.

### CODE REVIEW

File: Nos/Extensions/UIImage+File.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import UIKit
+import Logger
+
+extension UIImage {
+    
+    /// Initializes an image in memory from the file at the given URL.
+    convenience init?(fileURL: URL) {
```
Comment (martindsq) [inline Nos/Extensions/UIImage+File.swift:7]: Is this function used in some place? I couldn't find an usage by just reading the code in Github.

Comment (mplorentz) [reply Nos/Extensions/UIImage+File.swift]: Nope! I used it in an earlier version but it's dead code now, so I removed it. Nice catch!

**Review by martindsq — CHANGES_REQUESTED** [review]
This is working fine for me! I left a couple of changes I'd like to see addressed.

I'm a bit worried about the edge cases, those users trying to upload very long videos or very high quality videos, either when recording or when choosing from the photo library. For the first case, we could use the `videoMaximumDuration` property, for the later we could use the deprecated `videoExportPreset`, switching to PHPickerViewController, or implementing a compression step where we decide how to compress videos and images intended for being used in a social network.

I'm fine with opening a ticket for that, but I'd like to see that implemented some day.

File: Nos/Views/New Note/ImagePickerButton.swift [hunk 2]
```diff
@@ -110,17 +110,24 @@ struct ImagePickerButton<Label>: View where Label: View {
         )
         .sheet(isPresented: showImagePicker) {
```
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Views/New Note/ImagePickerButton.swift:111]: Good point. I changed it "Take photo or video" to keep it short.

**Comment by mplorentz** [comment]
@martindsq I'm glad you raised these concerns. I started a [discussion about them in Slack](https://planetary-app.slack.com/archives/CE6RE5400/p1712779079131649) and we can ticket any further work that comes out of that.

I also added a little code to pass through error messages from the Nostr.build API, so now if you try to upload a file that is too large it will look like this:
Screenshot 2024-04-10 at 3 52 42 PM

**Comment by martindsq** [comment]
@mplorentz I'm liking the direction this is taking! We are very close to wrap this up. Can you still add a guard at `upload(fileAt fileUrl: URL)`, before calling the `Data` initializer and checking if the file size of the file at the url exceeds some large value (definitely a larger value than the limit nostr.build has? Something like 100mb could work. That will prevent the app from trying to allocate so much space to a single address.

**Comment by mplorentz** [comment]
@martindsq I mentioned this in Slack but should have put it here too. The photo picker already limits videos to 10 minutes long. Based on my tests that's around 200mb, which doesn't put us in danger of running out of RAM.

**Review by martindsq — APPROVED** [review]

**Pull request merged**